generate/generate.js

```diff
@@ -0,0 +1,253 @@
+
+  const allFunctionsCode = Object.entries(extractedFunctions)
+    .map(([key, value]) => generateFunctionCode(key, value))
+    .join("\n\n");
+
+  const mainFile = `
+module ${exercise} exposing (${functionList})
+
+${allFunctionsCode}
+`;
+  console.log(mainFile);
+
+  const allTestCode = generateAllTestsCode(
+    exercise,
+    extractedFunctions,
+    canonicalData
+  );
+
+  const testFile = `
+module Tests exposing (tests)
+
+import ${exercise}
+import Expect
+import Test exposing (Test, describe, skip, test)
+
+${toElmComments(canonicalData.comments)}
+
+tests : Test
+tests = describe "${exercise}" [ ${allTestCode} ]
+`;
+
+  console.log(testFile);
+}
+
+// Generate the template code for one function
+function generateFunctionCode(name, { args, canError }) {
+  const returnType = canError ? "Result String Todo" : "Todo";
+  const typeAnnotation = [...Array(args.length).fill("Todo"), returnType].join(
+    " -> "
+  );
+  return `
+${name} : ${typeAnnotation}
+${name} ${args.join(" ")} = Debug.todo "Please implement ${name}"
+`;
+}
+
+// Generates the test code that should replace `<tests>` in
+// `describe "<exercise>" [ <tests> ]`
+function generateAllTestsCode(exercise, functions, { cases }) {
+  let generatedChunks = [];
+  for (const testCase of cases) {
+    if (testCase.hasOwnProperty("cases")) {
+      // Recursive call with the subgroup
+      const subtestsOutput = generateAllTestsCode(
+        exercise,
+        functions,
+        testCase
+      );
+      generatedChunks.push(
+        `
+  ${toElmComments(testCase.comments)}
+
+  describe "${testCase.description}" [ ${subtestsOutput} ]
+  `
+      );
+    } else {
+      generatedChunks.push(generateTestCode(exercise, functions, testCase));
+    }
+  }
+  return generatedChunks.join(", ");
+}
+
+// Generate the test code for one test case
+function generateTestCode(exercise, functions, testCase) {
+  // Extract the expected value
+  let expectedValue = testCase.expected;
+
+  // If the function can error, adjust the expected value
+  if (functions[testCase.property].canError) {
+    if (expectedValue.hasOwnProperty("error")) {
+      expectedValue = "Err " + jsonValueToElm(expectedValue.error);
+    } else {
+      expectedValue = "Ok " + jsonValueToElm(expectedValue);
+    }
+    // Otherwise just convert it into a string
+  } else {
+    expectedValue = jsonValueToElm(expectedValue);
+  }
+
+  // Stringify the test case inputs
+  let inputs = "";
+  for (const [key, value] of Object.entries(testCase.input)) {
+    inputs += " " + jsonValueToElm(value);
+  }
+
+  // Generate a re-implement message if this tests re-implements another one
+  let reimplements = [];
+  if (testCase.reimplements) {
+    reimplements = [
+      "This test reimplements the test with uuid " + testCase.reimplements,
+      "Please identify that test and remove it. Link:",
+      "https://github.com/exercism/problem-specifications/blob/main/exercises/" +
+        testCase.reimplements +
```

Comment:
This value here should be the original slug, not the uuid

Reply:
Woops, good catch